dev-qt/qt-creator/qt-creator-8.0.0.ebuild

```diff
@@ -0,0 +1,416 @@
+# USAGE: <flag> <subdir> <files...>
+# DESCRIPTION:
+# <flag> is the name of a flag in IUSE.
+# <subdir> is the  name of a directory called with add_subdirectory().
+# <files...> is a list of one or more qmake project files.
+#
+# This function patches <files> to remove add_subdirectory(<subdir>) from cmake
+# when <flag> is disabled, otherwise it does nothing. This can be useful to
+# avoid an automagic dependency when a subdirectory is added in cmake but the
+# corresponding feature USE flag is disabled. Similar to qt_use_disable_config()
+# from /qt5-build.eclass
+cmake_use_remove_addsubdirectory() {
+	[[ $# -ge 3 ]] || die "${FUNCNAME}() requires at least three arguments"
+	local flag=$1
+	local subdir=$2
+	shift 2
+
+	if ! use "${flag}"; then
+		echo "$@" | xargs sed -i -e "/add_subdirectory(${subdir})/d" || die
+	fi
+}
+
+llvm_check_deps() {
+	has_version -d "sys-devel/clang:${LLVM_SLOT}"
+}
+
+pkg_setup() {
+	llvm_pkg_setup
+	export CLANG_PREFIX="$(get_llvm_prefix ${LLVM_MAX_SLOT})"
+}
+
+src_prepare() {
+	cmake_src_prepare
+
+	# Remove automagic dep for qt5/qt6
+	sed -e "/^find_package(Qt6/,/else()/ s|if (NOT Qt6_FOUND)|if (1)|" \
+		-i cmake/FindQt5.cmake || die
+
+	# qt-creator hardcodes the CLANG_INCLUDE_DIR to the default.
+	# However, in sys-devel/clang, the directory changes with respect to
+	# -DCLANG_RESOURCE_DIR.  We sed in the correct include dir.
+	local res_dir="$(${CLANG_PREFIX}/bin/clang -print-resource-dir || die)"
+	sed -e "/\w*CLANG_INCLUDE_DIR=/s|=.*|=\"${res_dir}/include\"|" \
+		-i src/plugins/clangtools/CMakeLists.txt || die
+
+	cmake_use_remove_addsubdirectory glsl glsl src/libs/CMakeLists.txt
+	cmake_use_remove_addsubdirectory lsp languageserverprotocol \
+		src/libs/CMakeLists.txt tests/auto/CMakeLists.txt
+	cmake_use_remove_addsubdirectory qml advanceddockingsystem \
+		src/libs/CMakeLists.txt
+	cmake_use_remove_addsubdirectory modeling modelinglib \
+		src/libs/CMakeLists.txt
+	cmake_use_remove_addsubdirectory test test \
+		src/plugins/mcusupport/CMakeLists.txt
+
+	# PLUGIN_RECOMMENDS is treated like a hard-dependency
+	sed -i -e '/PLUGIN_RECOMMENDS /d' \
+		src/plugins/*/CMakeLists.txt || die
+
+	if use test; then
+		# No need to unbundle gtest since it links to unittests that won't be installed
+		find "${S}" -type f -name CMakeLists.txt -exec \
+			xargs sed -i -e 's|TARGET GoogleBenchmark|benchmark_FOUND|g' \
+				-e 's|GoogleBenchmark\( MODULE\)\?|benchmark|g' {} \; || die
+	else
+		rm -rf tests src/plugins/mcusupport/test || die
+	fi
+
+	# fix translations
+	local languages=()
+	for lang in ${PLOCALES}; do
+		use l10n_${lang} && languages+=( "${lang/-/_}" )
+	done
+	sed -i -e "s|^set(languages.*|set(languages ${languages[*]})|" \
+		share/qtcreator/translations/CMakeLists.txt || die
+
+	# remove bundled yaml-cpp
+	rm -r src/libs/3rdparty/yaml-cpp || die
+
+	# remove bundled qbs
+	rm -r src/shared/qbs || die
+
+	# remove bundled litehtml
+	rm -r src/libs/qlitehtml || die
```

Comment:
...and where is the unbundled dependency then? Without litehtml and without webengine, the documentation viewer will fall back to some very basic built-in viewer which really sucks (last time I checked at least).

Reply:
I don't have a problem leaving it bundled, but seeing as litehtml is its own project and different devs have different levels of tolerance for bundling, I would rather have waited for the OK from you.

Reply:
And about the plugin USE flags.  I have been back and forth on it more than I care to admit.  I would much prefer to get rid of them.  The end user can decide to enable/disable them at runtime.  I'd unconditionally build most plugins, but maybe keep some metaflags like `clang` and `qml`.  Either, `IUSE="clang debug doc qml systemd test tools wayland"`.  or `IUSE="debug doc test tools"`, or `IUSE="debug doc test"`. Let me know.

Reply:
> and since different devs have different levels of tolerance for bundling, I would rather have waited for the OK from you.

I'm not sure why you'd need an explicit OK from me since I've already made this decision long ago, the current in-tree ebuild already has the bundled litehtml. And generally speaking, Gentoo has always been fairly pragmatic in this regard. Unbundling makes sense only when we can provide a viable alternative, which isn't the case here. Removing features for the sake of unbundling is not the Gentoo way.

Reply:
Sorry.  For some reason I had assumed it was a recent addition and forgot that it was already bundled in 6.0, likely because it wasn't directly referenced in the ebuild.

Reply:
no worries